app/helpers/components_helper.rb

```diff
@@ -12,6 +12,10 @@ def maintaining_iterations_summary_table(iterations)
                     })
   end
 
+  def notification_icon(user)
```

Comment:
What do you think about rather than this behind a helper module, these being actual classes that are called? I dislike that we pollute the view context with some many magic helper methods. Feels brittle.

So we'd have a `app/view_components/` folder with `notification_icon.rb`, which inherits from `ViewComponent.rb` that has the `react_component` in.

```
Then we'd call `ViewComponents::NotificationIcon.render(user)`, which outputs the relevant haml.
```

That means it's a little less magic than the view helper methods, and it makes them easier to test functionally. The only thing I'm unclear of is the ease of using the view context, but I reckon it would be ok.

---

This wouldn't block this PR, but it could be a good refactor if y'all agreed?

Reply:
I like the idea. It's also okay to pass the view_context into the component as it makes it explicit what the component is rendered for. It also helps us look at all our components at a glance, since we have separate files for each.